tapeagents/tools/web_search.py

```diff
@@ -125,6 +156,26 @@ def execute_action(self, action: SearchAction) -> SearchResultsObservation:
         return SearchResultsObservation(query=action.query, serp=results, error=error)
 
 
+class SafeWebSearch(WebSearch):
+    llm: LLM | None = None
+
+    def execute_action(self, action: SafeSearchAction) -> SearchResultsObservation:
+        if action.source == "wiki":
+            query = f"site:wikipedia.org {action.query}"
+        elif action.source == "youtube":
+            query = f"site:youtube.com {action.query}"
+        else:
+            query = action.query
+        error = None
+        results = []
+        try:
+            results = safe_search(self.llm, query, private_context=action._private_context, max_results=5)
```

Comment:
we should return the rewritten new_query here and add it to the `SearchResultsObservation` few lines below instead of the original query